### PR TITLE
Prometheus 2.43 GMP support

### PR DESCRIPTION
**Pull request by TheSpiritXIII**
https://github.com/GoogleCloudPlatform/prometheus-engine/pull/562

### CODE REVIEW

File: cmd/prometheus/main_test.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ func TestAgentSuccessfulStartup(t *testing.T) {
 }
 
 func TestAgentFailedStartupWithServerFlag(t *testing.T) {
-	prom := exec.Command(promPath, "-test.main", "--enable-feature=agent", "--storage.tsdb.path=.", "--config.file="+promConfig)
+	prom := exec.Command(promPath, "-test.main", "--enable-feature=agent", "--storage.tsdb.path=.", "--config.file="+promConfig, "--export.debug.disable-auth")
```
Comment (bwplotka) [inline cmd/prometheus/main_test.go:378]: Now I remember why we created this option =D

**Review by bwplotka — APPROVED** [review]
LGTM,  went through all commits - nothing controversial 💪🏽 Thanks! Let's do it!

**Comment by TheSpiritXIII** [comment]
Note: We are getting ready to release prometheus-engine 0.9 which would include an export library bump to support this. When this is released, we can update and merge this PR.

**Comment by bwplotka** [comment]
NOTE: We need https://github.com/GoogleCloudPlatform/prometheus/pull/127 to fix MEDIUM vuln. here.

**Pull request merged**